Move more code to NodeInlines.h

rniwa opened the pull request:
#### a0b88c20c9152bcbfb261f0955797782f5924df3
<pre>
Move more code to NodeInlines.h
<a href="https://bugs.webkit.org/show_bug.cgi?id=292243">https://bugs.webkit.org/show_bug.cgi?id=292243</a>

Reviewed by Anne van Kesteren.

Moved more inline functions from Node.h to NodeInlines.h

* Source/WebCore/dom/Attr.cpp:
* Source/WebCore/dom/ContainerNode.cpp:
* Source/WebCore/dom/ContainerNodeAlgorithms.cpp:
* Source/WebCore/dom/Document.cpp:
* Source/WebCore/dom/Node.h:
(WebCore::Node::relaxAdoptionRequirement): Moved to NodeInlines.h.
(WebCore::Node::hasOneRef const): Ditto.
(WebCore::addSubresourceURL): Ditto.
(WebCore::Node::parentNodeGuaranteedHostFree const): Ditto.
(WebCore::Node::clearStyleFlags): Ditto.
(WebCore::Node::clearChildNeedsStyleRecalc): Ditto.
(WebCore::Node::setHasValidStyle): Ditto.
(WebCore::Node::setTreeScopeRecursively): Ditto.
(WebCore::Node::traverseToRootNodeInternal): Ditto.
* Source/WebCore/dom/NodeInlines.h:
(WebCore::Node::hasOneRef const): Moved from Node.h.
(WebCore::Node::clearStyleFlags): Ditto.
(WebCore::Node::clearChildNeedsStyleRecalc): Ditto.
(WebCore::Node::setHasValidStyle): Ditto.
(WebCore::Node::setTreeScopeRecursively): Ditto.
(WebCore::Node::parentNodeGuaranteedHostFree const): Ditto.
(WebCore::Node::traverseToRootNodeInternal): Ditto.
(WebCore::Node::relaxAdoptionRequirement): Ditto.
(WebCore::addSubresourceURL): Ditto.
* Source/WebCore/dom/ProcessingInstruction.cpp:
* Source/WebCore/html/HTMLBodyElement.cpp:
* Source/WebCore/html/HTMLEmbedElement.cpp:
* Source/WebCore/html/HTMLImageElement.cpp:
* Source/WebCore/html/HTMLInputElement.cpp:
* Source/WebCore/html/HTMLLinkElement.cpp:
* Source/WebCore/html/HTMLObjectElement.cpp:
* Source/WebCore/html/HTMLScriptElement.cpp:
* Source/WebCore/html/HTMLTableCellElement.cpp:
* Source/WebCore/html/HTMLTableElement.cpp:
* Source/WebCore/html/HTMLTemplateElement.cpp:
* Source/WebCore/html/track/TextTrackCue.cpp:
* Source/WebCore/page/DOMSelection.cpp:
* Source/WebCore/style/StyleTreeResolver.cpp:
* Source/WebCore/svg/SVGCursorElement.cpp:
* Source/WebCore/svg/SVGFEImageElement.cpp:
* Source/WebCore/svg/SVGImageElement.cpp:
* Source/WebCore/svg/SVGScriptElement.cpp:

Canonical link: <a href="https://commits.webkit.org/294302@main">https://commits.webkit.org/294302@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/dfbc7b358b25c329038e4c214fc63230e9d7cf7b

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/101363 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/21027 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/11332 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/106516 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/51992 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/103403 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/21335 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/29521 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/77209 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/34245 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/104370 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/16463 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/91557 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/57550 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/16283 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/9567 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/51340 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/86164 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/9643 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/108868 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/28492 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/20964 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/86184 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/28854 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/87759 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/85741 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/30465 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/8176 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/22609 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/16504 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/28423 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/33702 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/28234 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/31554 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/29793 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

annevk commented on Source/WebCore/dom/Node.h:
```diff
@@ -784,8 +777,6 @@ class Node : public EventTarget, public CanMakeCheckedPtr<Node> {
     bool m_adoptionIsRequired { true };
     bool m_deletionHasBegun { false };
-
-    friend void adopted(Node*);
```
This seems to have gotten dropped and I don't quite understand why.

rniwa replied:
oh, this was unused code!

rniwa replied:
Turns out this was not unused after all :( Debug builds rely on this to update `m_adoptionIsRequired`. Resurrected the function.

annevk reviewed (approved):
Looks good, modulo one part I don't quite understand, but I trust you do.

rniwa commented on Source/WebCore/dom/Node.h:
```diff
@@ -812,17 +803,6 @@ WEBCORE_EXPORT std::partial_ordering treeOrderForTesting(TreeType, const Node&,
 
-#if ASSERT_ENABLED
-
-inline void adopted(Node* node)
```
oh! this wasn't unused code after all. It's used by `Ref.h`.

rniwa committed:
Move more code to NodeInlines.h
https://bugs.webkit.org/show_bug.cgi?id=292243

Reviewed by Anne van Kesteren.

Moved more inline functions from Node.h to NodeInlines.h

* Source/WebCore/dom/Attr.cpp:
* Source/WebCore/dom/ContainerNode.cpp:
* Source/WebCore/dom/ContainerNodeAlgorithms.cpp:
* Source/WebCore/dom/Document.cpp:
* Source/WebCore/dom/Node.h:
(WebCore::Node::relaxAdoptionRequirement): Moved to NodeInlines.h.
(WebCore::Node::hasOneRef const): Ditto.
(WebCore::addSubresourceURL): Ditto.
(WebCore::Node::parentNodeGuaranteedHostFree const): Ditto.
(WebCore::Node::clearStyleFlags): Ditto.
(WebCore::Node::clearChildNeedsStyleRecalc): Ditto.
(WebCore::Node::setHasValidStyle): Ditto.
(WebCore::Node::setTreeScopeRecursively): Ditto.
(WebCore::Node::traverseToRootNodeInternal): Ditto.
* Source/WebCore/dom/NodeInlines.h:
(WebCore::Node::hasOneRef const): Moved from Node.h.
(WebCore::Node::clearStyleFlags): Ditto.
(WebCore::Node::clearChildNeedsStyleRecalc): Ditto.
(WebCore::Node::setHasValidStyle): Ditto.
(WebCore::Node::setTreeScopeRecursively): Ditto.
(WebCore::Node::parentNodeGuaranteedHostFree const): Ditto.
(WebCore::Node::traverseToRootNodeInternal): Ditto.
(WebCore::Node::relaxAdoptionRequirement): Ditto.
(WebCore::addSubresourceURL): Ditto.
* Source/WebCore/dom/ProcessingInstruction.cpp:
* Source/WebCore/html/HTMLBodyElement.cpp:
* Source/WebCore/html/HTMLEmbedElement.cpp:
* Source/WebCore/html/HTMLImageElement.cpp:
* Source/WebCore/html/HTMLInputElement.cpp:
* Source/WebCore/html/HTMLLinkElement.cpp:
* Source/WebCore/html/HTMLObjectElement.cpp:
* Source/WebCore/html/HTMLScriptElement.cpp:
* Source/WebCore/html/HTMLTableCellElement.cpp:
* Source/WebCore/html/HTMLTableElement.cpp:
* Source/WebCore/html/HTMLTemplateElement.cpp:
* Source/WebCore/html/track/TextTrackCue.cpp:
* Source/WebCore/page/DOMSelection.cpp:
* Source/WebCore/style/StyleTreeResolver.cpp:
* Source/WebCore/svg/SVGCursorElement.cpp:
* Source/WebCore/svg/SVGFEImageElement.cpp:
* Source/WebCore/svg/SVGImageElement.cpp:
* Source/WebCore/svg/SVGScriptElement.cpp:

Canonical link: https://commits.webkit.org/294302@main
